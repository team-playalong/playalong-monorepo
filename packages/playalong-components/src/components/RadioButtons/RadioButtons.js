import * as React from 'react';
import { arrayOf, shape, string, func } from 'prop-types';
import { RadioButton, RadioButtonGroup } from 'material-ui/RadioButton';
import MuiThemeProvider from 'material-ui/styles/MuiThemeProvider';
import THEME from '../../utils/theme';

function RadioButtons(props) {

  let currId;
  const handleChange = (e, val) => {
    props.onRadioChanged(val);
  };
  const allRadioButtons = props.inputs.map(input => {
    currId = input.value + Date.now();
    return (
        <RadioButton
          value={input.value}
          label={input.label}
          key={currId}
        />
    );
  });

  return (
    <MuiThemeProvider muiTheme={THEME}>
      <div className='ply-radio-buttons'>
        {!!props.legend && <label>{props.legend}</label>}
        <RadioButtonGroup
          onChange={handleChange}
          name={props.radioName || 'radio'}>
          {allRadioButtons}
        </RadioButtonGroup>
      </div>
    </MuiThemeProvider>
  );
};

const inputProp = shape({
  value: string,
  label: string,
});
RadioButtons.propTypes = {
  legend: string,
  inputs: arrayOf(inputProp),
  onRadioChanged: func,
  radioName: string,
};
export const props = ['inputs', 'onRadioChanged'];
export default RadioButtons;
