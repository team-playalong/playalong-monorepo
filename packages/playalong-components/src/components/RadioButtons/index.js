export {default, props} from './RadioButtons';
