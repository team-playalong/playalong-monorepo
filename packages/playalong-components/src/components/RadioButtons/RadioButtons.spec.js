import * as React from 'react';
import { shallow } from 'enzyme';
import renderer from 'react-test-renderer';

import RadioButtons from './RadioButtons';

let wrapper;
const props = {
  inputs: [
    { label: 'l1', value: 'v1' },
    { label: 'l2', value: 'v2' },
  ],
  radioName: 'myRadio',
};

beforeEach(() => {
  wrapper = shallow(<RadioButtons {...props} />);
});

test('Snapshot testing', () => {
  const tree = renderer.create(<RadioButtons
    inputs={[
      { label: 'l1', value: 'v1' },
      { label: 'l2', value: 'v2' },
    ]}
    radioName='myRadio' />).toJSON();

  expect(tree).toMatchSnapshot();
});
