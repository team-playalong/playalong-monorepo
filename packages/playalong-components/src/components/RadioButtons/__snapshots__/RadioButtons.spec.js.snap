// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Snapshot testing 1`] = `
<div
  className="ply-radio-buttons"
>
  <div
    className={undefined}
    style={
      Object {
        "muiPrepared": true,
      }
    }
  >
    <div
      className={undefined}
      style={
        Object {
          "cursor": "pointer",
          "display": "table",
          "height": "auto",
          "muiPrepared": true,
          "overflow": "visible",
          "position": "relative",
          "width": "100%",
        }
      }
    >
      <input
        checked={false}
        disabled={false}
        name="myRadio"
        onBlur={[Function]}
        onChange={[Function]}
        onFocus={[Function]}
        onMouseDown={[Function]}
        onMouseLeave={[Function]}
        onMouseUp={[Function]}
        onTouchEnd={[Function]}
        onTouchStart={[Function]}
        style={
          Object {
            "boxSizing": "border-box",
            "cursor": "inherit",
            "height": "100%",
            "left": 0,
            "margin": 0,
            "muiPrepared": true,
            "opacity": 0,
            "padding": 0,
            "pointerEvents": "all",
            "position": "absolute",
            "width": "100%",
            "zIndex": 2,
          }
        }
        type="radio"
        value="v1"
      />
      <div
        style={
          Object {
            "display": "flex",
            "height": "100%",
            "width": "100%",
          }
        }
      >
        <div
          style={
            Object {
              "display": "block",
              "flexShrink": 0,
              "float": "left",
              "height": 24,
              "marginLeft": 0,
              "marginRight": 16,
              "muiPrepared": true,
              "position": "relative",
              "transition": "all 450ms cubic-bezier(0.23, 1, 0.32, 1) 0ms",
              "width": 24,
            }
          }
        >
          <div>
            <svg
              onMouseEnter={[Function]}
              onMouseLeave={[Function]}
              style={
                Object {
                  "color": "#212121",
                  "display": "inline-block",
                  "fill": "#212121",
                  "height": 24,
                  "muiPrepared": true,
                  "opacity": 1,
                  "position": "absolute",
                  "transform": "scale(1)",
                  "transition": "all 450ms cubic-bezier(0.23, 1, 0.32, 1) 0ms",
                  "userSelect": "none",
                  "width": 24,
                }
              }
              viewBox="0 0 24 24"
            >
              <path
                d="M12 2C6.48 2 2 6.48 2 12s4.48 10 10 10 10-4.48 10-10S17.52 2 12 2zm0 18c-4.42 0-8-3.58-8-8s3.58-8 8-8 8 3.58 8 8-3.58 8-8 8z"
              />
            </svg>
            <svg
              onMouseEnter={[Function]}
              onMouseLeave={[Function]}
              style={
                Object {
                  "color": "#212121",
                  "display": "inline-block",
                  "fill": "#ec407a",
                  "height": 24,
                  "muiPrepared": true,
                  "opacity": 1,
                  "position": "absolute",
                  "transform": "scale(0)",
                  "transformOrigin": "50% 50%",
                  "transition": "all 450ms cubic-bezier(0.23, 1, 0.32, 1) 0ms",
                  "userSelect": "none",
                  "width": 24,
                }
              }
              viewBox="0 0 24 24"
            >
              <path
                d="M12 7c-2.76 0-5 2.24-5 5s2.24 5 5 5 5-2.24 5-5-2.24-5-5-5zm0-5C6.48 2 2 6.48 2 12s4.48 10 10 10 10-4.48 10-10S17.52 2 12 2zm0 18c-4.42 0-8-3.58-8-8s3.58-8 8-8 8 3.58 8 8-3.58 8-8 8z"
              />
            </svg>
          </div>
          <div
            onMouseDown={[Function]}
            onMouseLeave={[Function]}
            onMouseUp={[Function]}
            onTouchEnd={[Function]}
            onTouchStart={[Function]}
          />
          <div
            style={
              Object {
                "height": "100%",
                "left": 0,
                "muiPrepared": true,
                "overflow": "hidden",
                "position": "absolute",
                "top": 0,
                "width": "100%",
              }
            }
          />
        </div>
        <label
          style={
            Object {
              "color": "#212121",
              "display": "block",
              "float": "left",
              "fontFamily": "Roboto, sans-serif",
              "lineHeight": "24px",
              "muiPrepared": true,
              "position": "relative",
              "width": "calc(100% - 60px)",
            }
          }
        >
          l1
        </label>
      </div>
    </div>
    <div
      className={undefined}
      style={
        Object {
          "cursor": "pointer",
          "display": "table",
          "height": "auto",
          "muiPrepared": true,
          "overflow": "visible",
          "position": "relative",
          "width": "100%",
        }
      }
    >
      <input
        checked={false}
        disabled={false}
        name="myRadio"
        onBlur={[Function]}
        onChange={[Function]}
        onFocus={[Function]}
        onMouseDown={[Function]}
        onMouseLeave={[Function]}
        onMouseUp={[Function]}
        onTouchEnd={[Function]}
        onTouchStart={[Function]}
        style={
          Object {
            "boxSizing": "border-box",
            "cursor": "inherit",
            "height": "100%",
            "left": 0,
            "margin": 0,
            "muiPrepared": true,
            "opacity": 0,
            "padding": 0,
            "pointerEvents": "all",
            "position": "absolute",
            "width": "100%",
            "zIndex": 2,
          }
        }
        type="radio"
        value="v2"
      />
      <div
        style={
          Object {
            "display": "flex",
            "height": "100%",
            "width": "100%",
          }
        }
      >
        <div
          style={
            Object {
              "display": "block",
              "flexShrink": 0,
              "float": "left",
              "height": 24,
              "marginLeft": 0,
              "marginRight": 16,
              "muiPrepared": true,
              "position": "relative",
              "transition": "all 450ms cubic-bezier(0.23, 1, 0.32, 1) 0ms",
              "width": 24,
            }
          }
        >
          <div>
            <svg
              onMouseEnter={[Function]}
              onMouseLeave={[Function]}
              style={
                Object {
                  "color": "#212121",
                  "display": "inline-block",
                  "fill": "#212121",
                  "height": 24,
                  "muiPrepared": true,
                  "opacity": 1,
                  "position": "absolute",
                  "transform": "scale(1)",
                  "transition": "all 450ms cubic-bezier(0.23, 1, 0.32, 1) 0ms",
                  "userSelect": "none",
                  "width": 24,
                }
              }
              viewBox="0 0 24 24"
            >
              <path
                d="M12 2C6.48 2 2 6.48 2 12s4.48 10 10 10 10-4.48 10-10S17.52 2 12 2zm0 18c-4.42 0-8-3.58-8-8s3.58-8 8-8 8 3.58 8 8-3.58 8-8 8z"
              />
            </svg>
            <svg
              onMouseEnter={[Function]}
              onMouseLeave={[Function]}
              style={
                Object {
                  "color": "#212121",
                  "display": "inline-block",
                  "fill": "#ec407a",
                  "height": 24,
                  "muiPrepared": true,
                  "opacity": 1,
                  "position": "absolute",
                  "transform": "scale(0)",
                  "transformOrigin": "50% 50%",
                  "transition": "all 450ms cubic-bezier(0.23, 1, 0.32, 1) 0ms",
                  "userSelect": "none",
                  "width": 24,
                }
              }
              viewBox="0 0 24 24"
            >
              <path
                d="M12 7c-2.76 0-5 2.24-5 5s2.24 5 5 5 5-2.24 5-5-2.24-5-5-5zm0-5C6.48 2 2 6.48 2 12s4.48 10 10 10 10-4.48 10-10S17.52 2 12 2zm0 18c-4.42 0-8-3.58-8-8s3.58-8 8-8 8 3.58 8 8-3.58 8-8 8z"
              />
            </svg>
          </div>
          <div
            onMouseDown={[Function]}
            onMouseLeave={[Function]}
            onMouseUp={[Function]}
            onTouchEnd={[Function]}
            onTouchStart={[Function]}
          />
          <div
            style={
              Object {
                "height": "100%",
                "left": 0,
                "muiPrepared": true,
                "overflow": "hidden",
                "position": "absolute",
                "top": 0,
                "width": "100%",
              }
            }
          />
        </div>
        <label
          style={
            Object {
              "color": "#212121",
              "display": "block",
              "float": "left",
              "fontFamily": "Roboto, sans-serif",
              "lineHeight": "24px",
              "muiPrepared": true,
              "position": "relative",
              "width": "calc(100% - 60px)",
            }
          }
        >
          l2
        </label>
      </div>
    </div>
  </div>
</div>
`;
